lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+              'res <- revdep_check();',
+              'if (length(res) > 0) {',
+              ' revdep_check_summary(res);',
+              ' revdep_check_save_logs(res);',
+              ' q(status = 1, save = "no");',
+              '}',
+            ].join(' ')
+            sh.cmd "Rscript -e '#{revdep_script}'"
+          end
+
+        end
+
+        def after_failure
+          dump_logs
+          super
+        end
+
+        private
+
+        def needs_bioc?
+          config[:bioc_required] or not config[:bioc_packages].empty?
+        end
+
+        def packages_as_arg(packages)
+          quoted_pkgs = packages.collect{|p| "\"#{p}\""}
+          "c(#{quoted_pkgs.join(', ')})"
+        end
+
+        def as_r_boolean(bool)
+          bool ? "TRUE" : "FALSE"
+        end
+
+        def r_install(packages)
+          sh.echo "Installing R packages: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          sh.cmd "Rscript -e 'install.packages(#{pkg_arg}, " +
+                 "repos=\"#{config[:cran]}]\")'"
+        end
+
+        def r_github_install(packages)
+          setup_devtools
+          sh.echo "Installing R packages from github: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          install_script = [
+            'library("devtools")',
+            "options(repos = c(CRAN = \"#{config[:cran]}\"))",
+            "install_github(#{pkg_arg}, build_vignettes = FALSE)",
+          ].join('; ')
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+        
+        def r_binary_install(packages)
+          case config[:os]
+          when 'linux'
+            sh.echo "Installing *binary* R packages: #{packages.join(', ')}"
+            apt_install packages.collect{|p| "r-cran-#{p.downcase}"}
+          else
+            sh.echo "R binary packages not supported on #{config[:os]}, " +
+                    'falling back to source install'
+            r_install packages
+          end
+        end
+
+        def apt_install(packages)
+          pkg_arg = packages.join(' ')
+          sh.echo "Installing apt packages: #{packages.join(', ')}"
+          sh.cmd "sudo apt-get install #{pkg_arg}", retry: true
+        end
+
+        def bioc_install(packages)
+          if not needs_bioc?
+            return
+          end
+          setup_bioc
+          sh.echo "Installing bioc packages: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          install_script = [
+            "source(\"#{config[:bioc]}\")",
+            'options(repos=biocinstallRepos())',
+            "biocLite(#{pkg_arg})",
+          ].join('; ')
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+
+        def install_deps
+          setup_devtools
+          if not needs_bioc?
+            install_script = [
+              'library("devtools")',
+              "options(repos = c(CRAN = \"#{config[:cran]}\"))",
+              'install_deps(dependencies = TRUE)',
```

Comment:
@hadley -- ah, nice. done. that actually shortened most of my inline scripts (save the revdeps one, which needs several functions from devtools).

@eddelbuettel you can still skip it -- if you provide your own `install` step in the `.travis.yml`, you'll avoid calling this one, and `devtools` won't get installed. that said, i suspect you're the **only** person who prefers to explicitly list their dependencies instead of just calling `install_deps`. :smirk: